Source/monster.cpp

```diff
@@ -4679,36 +4678,34 @@ void PrintMonstHistory(int mt)
 			minHP = 4 * minHP + hpBonusHell;
 			maxHP = 4 * maxHP + hpBonusHell;
 		}
-		strcpy(tempstr, fmt::format(_("Hit Points: {:d}-{:d}"), minHP, maxHP).c_str());
-		AddPanelString(tempstr);
+		AddPanelString(fmt::format(_("Hit Points: {:d}-{:d}"), minHP, maxHP));
 	}
 	if (MonsterKillCounts[mt] >= 15) {
 		int res = (sgGameInitInfo.nDifficulty != DIFF_HELL) ? MonstersData[mt].mMagicRes : MonstersData[mt].mMagicRes2;
 		if ((res & (RESIST_MAGIC | RESIST_FIRE | RESIST_LIGHTNING | IMMUNE_MAGIC | IMMUNE_FIRE | IMMUNE_LIGHTNING)) == 0) {
-			strcpy(tempstr, _("No magic resistance"));
-			AddPanelString(tempstr);
+			AddPanelString(_("No magic resistance"));
 		} else {
 			if ((res & (RESIST_MAGIC | RESIST_FIRE | RESIST_LIGHTNING)) != 0) {
-				strcpy(tempstr, _("Resists: "));
+				std::string resists = _("Resists: ");
 				if ((res & RESIST_MAGIC) != 0)
-					strcat(tempstr, _("Magic "));
+					resists.append(_("Magic "));
 				if ((res & RESIST_FIRE) != 0)
-					strcat(tempstr, _("Fire "));
+					resists.append(_("Fire "));
 				if ((res & RESIST_LIGHTNING) != 0)
-					strcat(tempstr, _("Lightning "));
-				string_view str { tempstr };
-				str.remove_suffix(str.size() - FindLastUtf8Symbols(str));
+					resists.append(_("Lightning "));
+				string_view str { resists };
+				str.remove_suffix(str.length() - FindLastUtf8Symbols(str));

```

Comment:
This is to remove the trailing space right? Would be easier to use space as a prefix on the resist type strings:

``` cpp
std::string resists = _("Resists:");
if ((res & RES_MAGIC) != 0)
    resists.append(_(" Magic"));
...
AddPanelString(resists);
```
(picking this up because the call to `string_view::size()` was changed here but not on line 4709)

Reply:
Could even do `esists.append(" ").append(_("Magic"));` so not to generate new translation strings for this.

Reply:
... except that some languages do not sure regular spaces for separation :face_in_clouds:

Reply:
Yeah, I remember you telling me that for another translation string 😄 . Figured this way using the space as a prefix lets the translations drop it if they don't need word separators. I'm happy to do that refactor as another PR anyway, for this one the main concern is just making the `size()`/`length()` calls consistent.

Reply:
Fixed, also replaced a bunch of `strcpy()` with `CopyUTF8()`, hope it's not to distracting in the middle of a review :)